modules/apps/fragment/fragment-impl/src/main/java/com/liferay/fragment/internal/renderer/FragmentEntryFragmentRenderer.java

```diff
@@ -140,6 +156,35 @@ private FragmentEntryLink _getFragmentEntryLink(
 		return fragmentEntryLink;
 	}
 
+	private String _getModuleName(long fragmentEntryId, String js) {
+		JSPackage jsPackage = _npmResolver.getJSPackage();
+
+		String moduleName = "fragment/" + fragmentEntryId;
+
+		JSModule jsModule = _npmRegistry.getJSModule(
+			ModuleNameUtil.getModuleId(jsPackage, moduleName));
+
+		// TODO: refresh JSModule when JavaScript changes or else find a better
+		// way to maintain JSModules for fragment in sync with DB
+
+		if (jsModule == null) {
+			int x = js.indexOf("'");
+
+			int y = js.indexOf("'", x + 1);
```

Comment:
Could we go with something more descriptive than `x`/`y` here? (`start`/`end`)?

`x`/`y` sound like 2D coordinates.

Reply:
Looks like it is the usual portal pattern :roll_eyes: . But I will try to find another pattern...